web/src/filter.ts

```diff
@@ -1426,4 +1488,25 @@ export class Filter {
             !this.has_operand("is", "starred")
         );
     }
+
+    try_adjusting_for_moved_with_target(message?: Message): void {
+        if (this.is_adjusted_for_moved_with_target) {
+            return;
+        }
+
+        if (!message) {
+            const message_id = Number.parseInt(this.operands("with")[0]!, 10);
+            message = message_store.get(message_id);
+        }
+
+        if (!message) {
+            return;
+        }
+
+        const adjust_terms = Filter.adjusted_terms_if_moved(this._terms, message);
```

Comment:
`adjusted_terms` seems like the right grammar, since this is not a boolean value.